tests/ExampleSchema.hs

```diff
@@ -0,0 +1,304 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- | An example GraphQL schema, used in our end-to-end tests.
+--
+-- Based on the example schema given in the GraphQL spec. See
+-- <https://facebook.github.io/graphql/#sec-Validation>.
+--
+-- Here's the full schema:
+--
+-- @
+-- enum DogCommand { SIT, DOWN, HEEL }
+--
+-- type Dog implements Pet {
+--   name: String!
+--   nickname: String
+--   barkVolume: Int
+--   doesKnowCommand(dogCommand: DogCommand!): Boolean!
+--   isHousetrained(atOtherHomes: Boolean): Boolean!
+--   owner: Human
+-- }
+--
+-- interface Sentient {
+--   name: String!
+-- }
+--
+-- interface Pet {
+--   name: String!
+-- }
+--
+-- type Alien implements Sentient {
+--   name: String!
+--   homePlanet: String
+-- }
+--
+-- type Human implements Sentient {
+--   name: String!
+-- }
+--
+-- enum CatCommand { JUMP }
+--
+-- type Cat implements Pet {
+--   name: String!
+--   nickname: String
+--   doesKnowCommand(catCommand: CatCommand!): Boolean!
+--   meowVolume: Int
+-- }
+--
+-- union CatOrDog = Cat | Dog
+-- union DogOrHuman = Dog | Human
+-- union HumanOrAlien = Human | Alien
+-- @
+--
+-- Unlike the spec, we don't define a @QueryRoot@ type here, instead
+-- encouraging test modules to define their own as appropriate to their needs.
+--
+-- We'll repeat bits of the schema below, explaining how they translate into
+-- Haskell as we go.
+
+module ExampleSchema
+  ( DogCommand
+  , DogCommandEnum
+  , Dog
+  , Sentient
+  , Pet
+  , Alien
+  , Human
+  , CatCommand
+  , Cat
+  , CatOrDog
+  , DogOrHuman
+  , HumanOrAlien
+  ) where
+
+import Protolude hiding (Enum)
+
+import GraphQL.API
+  ( GraphQLEnum(..)
+  , Enum
+  , Object
+  , Field
+  , Argument
+  , Interface
+  , Union
+  , (:>)
+  )
+-- XXX: This really shouldn't be part of Resolver, since whether or not a
+-- thing has a default is part of the API / Schema definition.
+import GraphQL.Resolver (Defaultable(..))
+import GraphQL.Internal.AST (getNameText)
+
+-- | A command that can be given to a 'Dog'.
+--
+-- @
+-- enum DogCommand { SIT, DOWN, HEEL }
+-- @
+--
+-- To define this in Haskell we need to do three things:
+--
+--  1. Define a sum type with nullary constructors to represent the enum
+--     (here, 'DogCommandEnum')
+--  2. Make it an instance of 'GraphQLEnum'
+--  3. Wrap the sum type in 'Enum', e.g. @Enum "DogCommand" DogCommandEnum@
+--     so it can be placed in a schema.
+data DogCommandEnum = Sit | Down | Heel deriving (Show, Eq, Ord, Generic)
+
+instance GraphQLEnum DogCommandEnum
```

Comment:
Good call. Done. I've also removed the type alias completely, instead embedding the `Enum "DogCommand" DogCommand"` in the schema. Because it's only there once, it works, and reduces the confusion about which one I mean.